pkg/controller/account/account_controller.go

```diff
@@ -660,9 +660,41 @@ func (r *ReconcileAccount) initializeRegions(reqLogger logr.Logger, currentAcctI
 		return err
 	}
 
+	if !currentAcctInstance.IsBYOC() { // This is a normal account creation
+		reqLogger.Info("Normal account region initialization")
+		go r.asyncRegionInit(reqLogger, currentAcctInstance, creds, regionAMIs, castAWSRegionType(regionsEnabledInAccount.Regions))
+		return nil
+	}
+
+	accountClaim, acctClaimErr := r.getAccountClaim(currentAcctInstance)
+	if acctClaimErr != nil {
+		reqLogger.Info("Accountclaim not found")
+		return acctClaimErr
+	}
+
+	for _, wantedRegion := range accountClaim.Spec.Aws.Regions {
+		found := false
+		for _, enabledRegion := range regionsEnabledInAccount.Regions {
+			reqLogger.Info("Comparing regions", "wantedRegion", wantedRegion, "enabledRegion", enabledRegion)
+			if wantedRegion.Name == *enabledRegion.RegionName {
+				found = true
+			}
+		}
+		if !found {
+			err := utils.SetAccountStatus(
+				r.Client,
+				reqLogger,
+				currentAcctInstance,
+				fmt.Sprintf("AWS region %s is not support supported for AWS account %s", wantedRegion, currentAcctInstance.Name),
+				awsv1alpha1.AccountInitializingRegions,
+			)
+			return err
```

Comment:
This should be moved out of the for loop entirely.

Reply:
I don't understand how moving it outside the for loop is beneficial, all it would accomplish is make it less efficient.

Reply:
You're right.  I got the way this works backwards in my brain.  That's what I get for "just popping on and reviewing this real quick" after hours haha.